lib/internal/bootstrap/browser.js

```diff
@@ -71,6 +71,30 @@ defineOperation(globalThis, 'setTimeout', timers.setTimeout);
 defineReplacableAttribute(globalThis, 'performance',
                           require('perf_hooks').performance);
 
+const { queueMicrotask } = require('internal/process/task_queues');
+// https://html.spec.whatwg.org/#runtime-script-errors
+function reportError(error) {
+  if (arguments.length === 0) {
+    const { codes } = require('internal/errors');
+    throw new codes.ERR_MISSING_ARGS('error');
+  }
+  try {
+    if (!error?.stack) {
+      // The user called reportError with a non-error so we want to give them
+      // a way to get a stack trace. We do this by emitting a warning the user
+      // so the user can run node with `--trace-warnings` to get a stack tace.
+      process.emitWarning('reportError was called with a non-error.');
```

Comment:
hmm.... I'm not convinced the warning is necessary or all that useful. Not blocking however.

Reply:
I mostly thing we shouldn’t encourage users to error with things without a stack trace

Reply:
Does the browser warn on this?

Not every error condition needs a stack trace.

Reply:
> Not every error condition needs a stack trace.

I am pretty sure when debugging a warning I would __really__ want a  stack trace though note this is opt in with `--trace-warnings` right?

Reply:
The individual user’s desire there may be for a stack trace, but that’s not universal, or the language wouldn't let you throw something without one.

Reply:
@ljharb as far as I'm aware the language is not _yet_ aware of stack traces anyway? So the language is completely oblivious to the debugging experience (or the user) isn't it?

The only reason stuff like `Error` even exists is that language itself can throw stuff that subclasses it (like RangeError or TypeError) and as a possible extensibility point. The language is specific about this:

> Instances of Error objects are thrown as exceptions when runtime errors occur. The Error objects may also serve
as base objects for user-defined exception classes.

(exists since es3)

In any case aren't "what errors users get" and "what the language lets you do" here completely orthogonal since it's something the runtime and not the language typically does?

Reply:
No - users (should) expect to be able to throw and catch any kind of value, because every kind of value can be an error. Thus, a warning here on an otherwise-valid error will be surprising and confusing to developers who expect that explicitly passing a non-Error object to reportError will be calmly accepted.